src/components/ProjectsContextProvider.tsx

```diff
@@ -53,12 +69,110 @@ export const ProjectsContextProvider = ({
   )
 }
 
+/**
+ * We need some of the functionality of the ProjectsContextProvider in the web version
+ * but we can't perform file system operations in the browser,
+ * so most of the behavior of this machine is stubbed out.
+ */
 const ProjectsContextWeb = ({ children }: { children: React.ReactNode }) => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const clearImportSearchParams = useCallback(() => {
+    // Clear the search parameters related to the "Import file from URL" command
+    // or we'll never be able cancel or submit it.
+    searchParams.delete(CREATE_FILE_URL_PARAM)
+    searchParams.delete('code')
+    searchParams.delete('name')
+    searchParams.delete('units')
```

Comment:
Could we use an enum or type to define this?

I see that we have `CREATE_FILE_URL_PARAM` but it maybe worth defining a type that is the query parameter for share urls and we can iterate over those keys and have some definition.

In the future we could have different query parameter encodings and those TS definitions would help us know which ones are valid.